packages/astro/src/assets/fonts/providers.ts

```diff
@@ -1,6 +1,41 @@
+import { createRequire } from 'node:module';
+import type { AstroSettings } from '../../types/astro.js';
 import { adobe } from './providers/adobe.js';
+import { google } from './providers/google.js';
+import { local } from './providers/local.js';
+import type { FontProvider, ResolvedFontProvider } from './types.js';
+import { fileURLToPath } from 'node:url';
 
-/** TODO: */
+/** TODO: jsdoc */
 export const fontProviders = {
 	adobe,
+	// TODO: reexport all unifont providers
 };
+
+function resolveEntrypoint(settings: AstroSettings, entrypoint: string): string {
+	const require = createRequire(settings.config.root);
+
+	try {
+		return require.resolve(entrypoint);
+	} catch {
+		return fileURLToPath(new URL(entrypoint, settings.config.root));
+	}
+}
```

Comment:
I don't think this is the correct approach. In dev, we need to use the `ModuleLoader` provided by `vite`, while during the build we need to rely on  vite plugins. 

We use this approach for various things.

Renderers
https://github.com/withastro/astro/blob/d1384b87e8e8eebe43058f2e29b16344637df7bf/packages/astro/src/core/render/renderer.ts#L5-L17

Middleware
https://github.com/withastro/astro/blob/d1384b87e8e8eebe43058f2e29b16344637df7bf/packages/astro/src/core/middleware/loadMiddleware.ts#L11-L18

Assets
https://github.com/withastro/astro/blob/d1384b87e8e8eebe43058f2e29b16344637df7bf/packages/astro/src/content/vite-plugin-content-assets.ts#L79